.buildkite/pipeline.yml

```diff
@@ -9,24 +9,18 @@ steps:
     if: &not_scheduled_autodoc
       build.branch != "autodoc/master" && build.branch != "master" && build.source != "schedule"
     commands:
-    - nix run -f ci.nix pkgs.hlint -c
+    - nix shell .#haskellPackages.hlint -c
         ./scripts/lint.sh
 
-  - label: reuse lint
+  - label: check
     if: *not_scheduled_autodoc
     commands:
-    - nix run -f ci.nix pkgs.reuse -c
-        reuse lint
-
-  - label: check trailing whitespace
-    if: *not_scheduled_autodoc
-    commands:
-    - .buildkite/check-trailing-whitespace.sh
+    - nix flake check -L
 
   - label: xrefcheck
     if: *not_scheduled_autodoc
     commands:
-    - nix run -f ci.nix pkgs.xrefcheck -c xrefcheck
+    - nix shell .#xrefcheck --impure -c xrefcheck
```

Comment:
I found out that `--impure` is required because xrefcheck isn't flakified or somesuch, but I have a feeling this `--impure` will sit here forever. Maybe add a todo to remove it?

Reply:
Xrefcheck is currently flakified, but looks like we won't have any chances to update this in future